src/os_posix/os_inmemory.c

```diff
@@ -0,0 +1,463 @@
+/*-
```

Comment:
It's providing OS-level functionality, though, for example, "does this file exist"?

I've tried to resist wholesale file movements until the Windows port is done, but we do need to do something.

There's going to be general OS/library functions, file-handle API functions, file-system API functions, and those 3 categories come in at least 3 flavors: Windows, POSIX, in-memory, and of course, some files are shared between categories.

Support is starting to be a little bit of a dumping ground (for example, the PowerPC checksum code is there, because the POSIX code was there, and then it was #ifdef'd for the x86).

So far I haven't thought of anything I like, better ideas welcomed.